Source/WebCore/html/HTMLCanvasElement.cpp

```diff
@@ -730,14 +738,24 @@ ExceptionOr<void> HTMLCanvasElement::toBlob(Ref<BlobCallback>&& callback, const
 
     auto encodingMIMEType = toEncodingMimeType(mimeType);
     auto quality = qualityFromJSValue(qualityValue);
-
-#if USE(CG)
-    if (auto imageData = getImageData()) {
+    auto scheduleCallbackWithBlobData = [&](Ref<BlobCallback>&& callback, Vector<uint8_t>&& blobData) {
         RefPtr<Blob> blob;
-        Vector<uint8_t> blobData = encodeData(imageData->pixelBuffer(), encodingMIMEType, quality);
         if (!blobData.isEmpty())
             blob = Blob::create(&document(), WTFMove(blobData), encodingMIMEType);
         callback->scheduleCallback(document(), WTFMove(blob));
+    };
+
+    if (document().requiresScriptExecutionTelemetry(ScriptTelemetryCategory::Canvas)) {
+        if (RefPtr buffer = createImageForNoiseInjection())
+            scheduleCallbackWithBlobData(WTFMove(callback), buffer->toData(encodingMIMEType, quality));
+        else
+            callback->scheduleCallback(document(), nullptr);
```

Comment:
Optional but I suppose you could combine the calls?

```suggestion
        RefPtr buffer = createImageForNoiseInjection();
        scheduleCallbackWithBlobData(WTFMove(callback), buffer ? buffer->toData(encodingMIMEType, quality) : Vector<uint8_t> { });
```

Reply:
Yep! Bit cleaner when combined.